source/base/parameter_handler.cc

```diff
@@ -519,17 +519,36 @@ ParameterHandler::parse_input(const std::string &filename,
       std::istream is(&fb);
       std::string file_ending = filename.substr(filename.find_last_of('.') + 1);
       boost::algorithm::to_lower(file_ending);
+
       if (file_ending == "prm")
         parse_input(is, filename, last_line, skip_undefined);
       else if (file_ending == "xml")
         parse_input_from_xml(is, skip_undefined);
       else if (file_ending == "json")
         parse_input_from_json(is, skip_undefined);
       else
-        AssertThrow(
-          false,
-          ExcMessage(
-            "Unknown input file. Supported types are .prm, .xml, and .json."));
+        {
+          // See: discussion in https://github.com/dealii/dealii/pull/9741
+          // This PR contains the code which we should use from release
+          // 9.3 onwards. File endings other than prm, xml, json are currently
+          // interpreted as files of prm-type for reasons of backward
+          // compatibility.
+#ifdef DEBUG
+          std::cout
+            << "WARNING: From release 9.3 onwards, this function will only accept files ending with .prm, .xml, or .json as parameter files."
+            << std::endl;
```

Comment:
Are we fine with printing this on all MPI ranks? It probably makes little sense to test for MPI ranks (as we do not know the communicator), so I guess a user cannot complain when getting thousands of those...

Reply:
> I guess a user cannot complain when getting thousands of those...

I hope the users don't run the code in debug mode with thousands of processes ;) But I am happy to change it if there are any suggestion!

Reply:
I would appreciate if there is a way to silence the warning at least.
Also, I would not mention any specific version at all. Let's just say it's deprecated.

Reply:
> I would appreciate if there is a way to silence the warning at least.

In the case of gcc, we could write `#ifdef __DEPRECATED` (https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html). But I am afraid that other compilers have other macros.

> Also, I would not mention any specific version at all. Let's just say it's deprecated.

I will do that!

Reply:
Would `#ifdef DEAL_II_DEPRECATED` work?

Reply:
No, we use that already as a replacement for `[[deprecated]]`. :slightly_smiling_face: `DEAL_II_ALLOW_DEPRECATED_BEHAVIOR` or something similar would work for me but others might have better suggestions.